### PR TITLE
Make Workflows Fetch Version Automatically

### PR DESCRIPTION
**Pull request by mingshl**
### Description

Make workflows fetch version automatically to reduce the number of files changed in every version bump. 

in  remote-integ-tests workflow: 
change to read version number from ./package.json file and read the branch name from GITHUB_REF
make remote-integ-tests-workflow no longer depends on other workflows
Change checkout opensearch based on search-processor repo instead of anomaly-detection.

in test and build workflow: 
change to read version number from ./package.json file and read the branch name from GITHUB_REF

### Issues Resolved
https://github.com/opensearch-project/dashboards-search-relevance/issues/140

### CODE REVIEW

File: .github/workflows/remote-integ-tests-workflow.yml [hunk 1]
```diff
@@ -34,23 +29,33 @@ jobs:
           distribution: 'corretto'
           java-version: '11'
 
+      - uses: actions/checkout@v2
+      - name: Fetch Version Information
+        run: |
+          OPENSEARCH_VERSION=$(node -p "require('./package.json').version")
+          BRANCH_NAME=${GITHUB_REF#refs/heads/}
```
Comment (msfroh) [inline .github/workflows/remote-integ-tests-workflow.yml:36]: This doesn't seem to work on PRs, since the refs/heads seems to point to the PR's ref.

Comment (mingshl) [reply .github/workflows/remote-integ-tests-workflow.yml]: right, so I modified the logic by 
if it's a pull request, get the branch name by the pull_request.base,
if it's push, get the branch name by the github.ref, 
it seems working from the log here https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/4887272941/jobs/8723655835?pr=177#step:3:16

**Review by msfroh — APPROVED** [review]

**Comment by noCharger** [comment]
Looks like this line not changed https://github.com/opensearch-project/dashboards-search-relevance/blob/2.x/.github/workflows/test-and-build.yml#LL8C1-L9C1

@mingshl can we have a fast follow-up?

**Comment by mingshl** [comment]
> Looks like this line not changed https://github.com/opensearch-project/dashboards-search-relevance/blob/2.x/.github/workflows/test-and-build.yml#LL8C1-L9C1
> 
> @mingshl can we have a fast follow-up?

Yes, because it was not backported to 2.x yet. If #177 is backported, the workflow will read from package.json file https://github.com/opensearch-project/dashboards-search-relevance/blob/bde3f40816f2add85cd785fada1299a598d2c9a5/.github/workflows/test-and-build.yml#L20

**Comment by noCharger** [comment]
@mingshl Looks like the github workflow remote-integ-tests-workflow was present and is disappered in this PR by looking at the commit history on [main ](https://github.com/opensearch-project/dashboards-search-relevance/commits/main) branch. And [build ](https://github.com/opensearch-project/dashboards-search-relevance/actions/runs/4931735507/jobs/8814075734)starts to fail since this PR

Can you check it out?

**Pull request merged**